From a6b33296edd901f5dc0e3a5b202d2676e0ea35da Mon Sep 17 00:00:00 2001
From: Zexin Fu <zexifu@iis.ee.ethz.ch>
Date: Tue, 31 Oct 2023 17:22:50 +0100
Subject: [PATCH] change the mode of stream_arbiter from rr to prio

---
 src/reg_mux.sv | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/reg_mux.sv b/src/reg_mux.sv
index 74f9806..80ddab2 100644
--- a/src/reg_mux.sv
+++ b/src/reg_mux.sv
@@ -50,7 +50,8 @@ module reg_mux #(
 
   stream_arbiter #(
     .DATA_T (req_payload_t),
-    .N_INP (NoPorts)
+    .N_INP (NoPorts),
+    .ARBITER ("prio")
   ) i_stream_arbiter (
     .clk_i,
     .rst_ni,
-- 
2.28.0

