From c6c590c4bda8b15ab1eef600272c32995dcbedef Mon Sep 17 00:00:00 2001
From: Nils Wistoff <nwistoff@iis.ee.ethz.ch>
Date: Tue, 13 Dec 2022 15:44:12 +0100
Subject: [PATCH 09/10] tb: Fix elf reading

Signed-off-by: Nils Wistoff <nwistoff@iis.ee.ethz.ch>
---
 smart_run/logical/tb/tb.v | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/smart_run/logical/tb/tb.v b/smart_run/logical/tb/tb.v
index 42c0d5c..312f14c 100644
--- a/smart_run/logical/tb/tb.v
+++ b/smart_run/logical/tb/tb.v
@@ -146,9 +146,9 @@ module tb();
     end
   
     $display("\t********* Read program *********");
-    void'(uvcl.get_arg_value("+ELF=", binary));
-    void'(read_elf(binary));
+    void'(uvcl.get_arg_value("++ELF=", binary));
     $display("Loading ELF binary %s", binary);
+    void'(read_elf(binary));
   
     $display("\t********* Load program to memory *********");
     while (get_section(section_addr, section_len)) begin
-- 
2.16.5

