From 14ad9afbb9d13f19c34344b3c2005511bb147edf Mon Sep 17 00:00:00 2001
From: Michael Rogenmoser <michael@rogenmoser.us>
Date: Thu, 5 Jan 2023 17:39:17 +0100
Subject: [PATCH] Select appropriate address range

---
 reg_top.sv.tpl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/reg_top.sv.tpl b/reg_top.sv.tpl
index 664e12801..56ff205c4 100644
--- a/reg_top.sv.tpl
+++ b/reg_top.sv.tpl
@@ -116,7 +116,7 @@ module ${mod_name} \
   // register signals
   logic           reg_we;
   logic           reg_re;
-  logic [AW-1:0]  reg_addr;
+  logic [BlockAw-1:0]  reg_addr;
   logic [DW-1:0]  reg_wdata;
   logic [DBW-1:0] reg_be;
   logic [DW-1:0]  reg_rdata;
@@ -316,7 +316,7 @@ module ${mod_name} \
 % if use_reg_iface:
   assign reg_we = reg_intf_req.valid & reg_intf_req.write;
   assign reg_re = reg_intf_req.valid & ~reg_intf_req.write;
-  assign reg_addr = reg_intf_req.addr;
+  assign reg_addr = reg_intf_req.addr[BlockAw-1:0];
   assign reg_wdata = reg_intf_req.wdata;
   assign reg_be = reg_intf_req.wstrb;
   assign reg_intf_rsp.rdata = reg_rdata;
-- 
2.31.1

